src/components/organisms/ConceptPicker.tsx

```diff
@@ -0,0 +1,167 @@
+  if (sort === "A-Z") {
+    return a.preferred_label.localeCompare(b.preferred_label);
+  } else if (sort === "Grouped") {
+    return a.recursive_subconcept_of[0].localeCompare(b.recursive_subconcept_of[0]);
+  }
+  return 0;
+};
+
+const filterConcepts = (concepts: TConcept[], search: string) => {
+  return concepts.filter(
+    (concept) =>
+      concept.preferred_label.toLowerCase().includes(search.toLowerCase()) ||
+      concept.alternative_labels.some((label) => label.toLowerCase().includes(search.toLowerCase()))
+  );
+};
+
+const onConceptChange = (router: NextRouter, concept: TConcept) => {
+  const query = { ...router.query };
+  let selectedConcepts: string | string[] = query[QUERY_PARAMS.concept_name] || [];
+
+  if (Array.isArray(selectedConcepts)) {
+    if (selectedConcepts.includes(concept.preferred_label)) {
+      selectedConcepts = selectedConcepts.filter((c) => c !== concept.preferred_label);
+    } else {
+      selectedConcepts = [...selectedConcepts, concept.preferred_label];
+    }
+  } else {
+    if (selectedConcepts === concept.preferred_label) {
+      selectedConcepts = [];
+    } else {
+      selectedConcepts = [selectedConcepts, concept.preferred_label];
+    }
+  }
+
+  query[QUERY_PARAMS.concept_name] = selectedConcepts;
+
+  router.push({ query: query }, undefined, { shallow: true });
+};
+
+const getSelectedConcepts = (concepts: TConcept[], router: NextRouter) => {
+  return concepts.reduce((acc, concept) => {
+    const queryValue = router.query[QUERY_PARAMS.concept_name];
+    if (!queryValue) {
+      return acc;
+    }
+    if (typeof queryValue === "string") {
+      return queryValue === concept.preferred_label ? [...acc, concept] : acc;
+    }
+    return queryValue.includes(concept.preferred_label) ? [...acc, concept] : acc;
+  }, [] as TConcept[]);
+};
+
+export const ConceptPicker = ({ concepts, startingSort = "A-Z" }: TProps) => {
+  const router = useRouter();
+  const ref = useRef(null);
+  const [search, setSearch] = useState("");
+  const [sort, setSort] = useState<TSort>(startingSort);
+  const [rootConcepts, setRootConcepts] = useState<TConcept[]>([]);
+  const [conceptsGrouped, setConceptsGrouped] = useState<{ [rootConceptId: string]: TConcept[] }>({});
+  const [filteredConcepts, setFilteredConcepts] = useState<TConcept[]>([]);
+
+  useEffect(() => {
+    const conceptIds = concepts.map((concept) => concept.wikibase_id);
+    fetchAndProcessConcepts(conceptIds).then(({ rootConcepts, concepts }) => {
+      setRootConcepts(rootConcepts);
+      setFilteredConcepts(concepts);
+      setConceptsGrouped(groupByRootConcept(concepts, rootConcepts));
+    });
+  }, [concepts]);
+
+  return (
+    <div className="relative flex flex-col gap-5 max-h-full pb-5" ref={ref}>
+      <div className="flex items-center justify-between">
+        <div className="flex items-center gap-2">
+          <div className="text-[15px] font-medium text-text-primary">Concepts</div>
```

Comment:
Note for later: As Conor's designs are now using 15px (which I've been avoiding thus far), we should get 15px in as a Tailwind text size. This will involve a noisy PR refactoring all of the existing text sizes, so let's park this for now.